### PR TITLE
fix: Oracle support for decimals

### LINKED ISSUE #517: Decimal values converted to Integer for oracle source
When we tested the max and sum aggregation functions on the decimal field from oracle. The data is truncated to integer type and hence the validation results are failing.

https://screenshot.googleplex.com/BdG5fXEgvvx4RaV

### LINKED ISSUE #520: Oracle to Bigquery Schema Validation Failure
When doing a schema validation check on Oracle to Bigquery field results in an error. 'Unsupported operand type'

The data type in oracle is NUMER(38) and Bigquery is BIGNUMERIC.

Seems like NUMBER(38) is getting converted to float and BIGNUMERIC is getting converted to Decimal.

Attached error and internal converted data type screenshot 
https://screenshot.googleplex.com/5C3Y6HNFLRMF6Em
https://screenshot.googleplex.com/9ZRAiAYsE88bFny

### PR DESCRIPTION
**Pull request by nehanene15**
Adds support for Oracle NUMBER() data type and treat is as a Decimal
Closes #517 
Closes #520

### CODE REVIEW

File: third_party/ibis/ibis_oracle/client.py [hunk 1]
```diff
@@ -12,21 +12,60 @@
...
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.NCLOB)
+def sa_oracle_NCLOB(_, satype, nullable=True):
+    return dt.String(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.LONG)
+def sa_oracle_LONG(_, satype, nullable=True):
+    return dt11.String(nullable=nullable)
```
Comment (sramsrinivasan) [inline third_party/ibis/ibis_oracle/client.py]: @nehanene15  - this should be dt.String?

File: third_party/ibis/ibis_oracle/client.py [hunk 2]
```diff
@@ -12,21 +12,60 @@
...
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.DATE)
+def sa_oracle_DATE(_, satype, nullable=True):
+    return dt.Date(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.VARCHAR)
+def sa_oracle_VARCHAR(_, satype, nullable=True):
+    return dt.String(nullable=nullable)
+
```
Comment (sramsrinivasan) [inline third_party/ibis/ibis_oracle/client.py:67]: @nehanene15  - Will it be a good idea to include all the below data types too or whatever you have here is sufficient?

from sqlalchemy.dialects.oracle import \
            BFILE, BLOB, CHAR, CLOB, DATE, \
            DOUBLE_PRECISION, FLOAT, INTERVAL, LONG, NCLOB, NCHAR, \
            NUMBER, NVARCHAR, NVARCHAR2, RAW, TIMESTAMP, VARCHAR, \
            VARCHAR2
https://docs.sqlalchemy.org/en/14/dialects/oracle.html#oracle-data-types

Comment (nehanene15) [reply third_party/ibis/ibis_oracle/client.py:67]: Good point - I'll add TIMESTAMP and VARCHAR2 since those will probably be the most common

**Review by sramsrinivasan — APPROVED** [review]

**Pull request merged**